internal/schemautil/user_config/service_types.go

```diff
@@ -0,0 +1,1419 @@
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"project_to_fork_from": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+		"service_to_fork_from": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+	}
+
+	return &schema.Schema{
+		Description:      "Clickhouse user configurable settings",
+		DiffSuppressFunc: schemautil.EmptyObjectDiffSuppressFuncSkipArrays(s),
+		Elem:             &schema.Resource{Schema: s},
+		MaxItems:         1,
+		Optional:         true,
+		Type:             schema.TypeList,
+	}
+}
+
+// ServiceTypeElasticsearch is a generated function returning the schema of the elasticsearch ServiceType.
+func ServiceTypeElasticsearch() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"custom_domain": {
+			Description: "Serve the web frontend using a custom CNAME pointing to the Aiven DNS name",
+			Optional:    true,
+			Required:    false,
+			Type:        schema.TypeString,
+		},
+		"disable_replication_factor_adjustment": {
+			Description: "DEPRECATED: Disable automatic replication factor adjustment for multi-node services. By default, Aiven ensures all indexes are replicated at least to two nodes. Note: Due to potential data loss in case of losing a service node, this setting can no longer be activated.",
+			Optional:    true,
+			Required:    false,
+			Type:        schema.TypeBool,
+		},
+		"elasticsearch": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"elasticsearch_version": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+		"index_patterns": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"index_template": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"ip_filter_map": {
+			Description: "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeMap,
+		},
+		"ip_filter_string": {
+			Description: "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"keep_index_refresh_interval": {
+			Description: "Aiven automation resets index.refresh_interval to default value for every index to be sure that indices are always visible to search. If it doesn't fit your case, you can disable this by setting up this flag to true.",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeBool,
+		},
+		"kibana": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"max_index_count": {
+			Description: "Maximum number of indexes to keep before deleting the oldest one",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeInt,
+		},
+		"opensearch_version": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
```

Comment:
We should have a validation to the fields that have it.

Reply:
could you please elaborate?